### ISSUE TITLE
Include better error message when extending from Equatable

### ISSUE THREAD

**Issue by alestiago** [isar/isar#1463]
**Is your feature request related to a problem? Please describe.**
Whenever a model extends a class from an external package it is not possible to add `@ignore`. For example, when extending from `Equatable`, it even has its own dedicated discussion https://github.com/isar/isar/discussions/648.

When running the isar_generator the error messages are quite handy but suggesting to add `@Collection(inheritance: false)` is not included within the error message. 

**Describe the solution you'd like**
As a developer, I would like the isar_generator to inform me to include `@Collection(inheritance: false)` whenever I'm inheriting from an external library. 

**Version**
 - Platform: N/A
 - Flutter version: 3.13.7
 - Isar version: ^3.1.0+1

**Comment by btrautmann** [comment]
100% agree--I only was able to resolve my issue after finding this and reading the linked discussion above (which indicated I needed _both_ `@ignore` _and_ `@Collection(inheritance: false)`, which I did not expect!

**Comment by vicenterusso** [comment]
I tried many times mixing `@collection` with other packages, including `freeze` and `equatable`. The only one that works without any issues is `dart_mappable`, which also has equality.

I have not tried to use with `@ignore` though

**Comment by btrautmann** [comment]
Yeah you'll need `@Collection` (capital C) with `inheritance: false` _and_ `@ignore` on the members you want `Isar` to ignore before it'll work. Took a lot of fiddling until I fully read the above discussion to actually get it. I'm now able to use `Equatable` which has been huge because I have a lot of `Riverpod` `Provider`s that expose `Stream`s of `Isar` objects and they were emitting redundant `List`s of info. With `Equatable` I'm able to do something like:

```dart
yield* CombineLatestStream.combine2(budgets, currencyFormats, (bs, cfs) {
    return bs.map((e) => e.toBudget(cfs.firstWhereOrNull((c) => c.budgetId == e.uuid))).toList();
  }).distinct((a, b) { // Use `distinct` to avoid emitting duplicate data
    return const DeepCollectionEquality.unordered().equals(a, b);
  });
```
Which avoids any flickering in the UI going from `Loaded` to `Loading` to `Loaded` state for no reason 👍
